### ISSUE TITLE
execbuilder: an internal error with INSERT and ORDER BY

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#57434]
These queries
```
CREATE TABLE target (x INT PRIMARY KEY, y INT, z INT, UNIQUE (y, z));
CREATE TABLE source (a INT, b INT, c INT);
INSERT INTO source VALUES (1, 1, 2), (1, 2, 1);
INSERT INTO target SELECT * FROM source ORDER BY rowid ON CONFLICT DO NOTHING;
```
result in
```
ERROR: internal error: column 9 not in input
SQLSTATE: XX000
DETAIL: stack trace:
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:105: getNodeColumnOrdinal()
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:666: func1()
github.com/cockroachdb/cockroach/pkg/sql/opt/colset.go:55: func1()
github.com/cockroachdb/cockroach/pkg/util/fast_int_set.go:165: ForEach()
github.com/cockroachdb/cockroach/pkg/sql/opt/colset.go:55: ForEach()
github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:664: applySimpleProject()
...
```

cc @RaduBerinde

**Comment by RaduBerinde** [comment]
The ORDER BY is the important thing, even though we are effectively ignoring it, that column is leftover somewhere.

**Comment by RaduBerinde** [comment]
This affects v20.2 as well (and maybe earlier ones).

**Comment by RaduBerinde** [comment]
It looks like this was fixed by #58679. I will add a regression test.
